client/main.go

```diff
@@ -160,41 +159,39 @@ func handleLeaveRoom(reader *bufio.Reader, state *State) {
 }
 
 func getRooms(w *bufio.Writer) {
-	sendRequest(w, scs.GET_ROOMS, nil)
+	sendRequest(w, scs.GetRooms, nil)
 }
 
 func joinRoom(w *bufio.Writer, name string) {
-	sendRequest(w, scs.JOIN_ROOM, scs.Room{Name: name})
+	sendRequest(w, scs.JoinRoom, scs.Room{Name: name})
 }
 
 func leaveRoom(w *bufio.Writer, name string) {
-	sendRequest(w, scs.LEAVE_ROOM, scs.Room{Name: name})
+	sendRequest(w, scs.LeaveRoom, scs.Room{Name: name})
 }
 
 func post(s *State) UserHandleFunc {
 	return func(w *bufio.Writer, msg string) {
-		sendRequest(w, scs.POST, scs.Message{Message: msg, Room: s.room.Name})
+		sendRequest(w, scs.Post, scs.Message{Message: msg, Room: s.room.Name})
 	}
 }
 
 func createRoom(w *bufio.Writer, name string) {
-	sendRequest(w, scs.CREATE_ROOM, scs.Room{Name: name})
+	sendRequest(w, scs.CreateRoom, scs.Room{Name: name})
 }
 
 func sendRequest(writer *bufio.Writer, actionName scs.ActionName, o interface{}) {
```

Comment:
Confirming this is the spot where it makes the most sense to handle errors, rather than returning them. I don't think callers would do anything differently than what this does, so it probably makes sense for the buck to stop here. If this were a public API, I'd probably say to return an error, though, so you don't have to make a breaking change later if one day you do need to return instead of log (in general libraries shouldn't log unless they're given a logger).

Reply:
I handled errors here because it was convenient, didn't have to repeat code, and don't have caller-specific ways to handle such errors. Yes I agree on all fronts.